OTAnalytics/plugin_ui/main_application.py

```diff
@@ -456,6 +463,35 @@ def _create_track_state(self) -> TrackState:
     def _create_track_view_state(self) -> TrackViewState:
         return TrackViewState()
 
+    def _wrap_plotter_with_cache(
+        self,
+        other: Plotter,
+        datastore: Datastore,
+        tracks: bool,
+        sections: bool,
+        flows: bool,
+    ) -> Plotter:
+        cached_plotter: CachedPlotter = CachedPlotter(other, subjects=[])
```

Comment:
We also need to invalidate the cache on event repository changes.
This would require modyfying the even repository use cases as subjects, so that we are able to register observers.

Reply:
Should be fixed now.